lib/string_reverse.rb

```diff
@@ -1,4 +1,19 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+  if my_string != nil
+    i = 0
+    j = my_string.length - 1
+
+    while i < j
+      x = my_string[i]
+      my_string[i] = my_string[j]
+      my_string[j] = x
+      i += 1
+      j -= 1
+    end
+
+    return my_string

```

Comment:
Minor: since you're reversing the string in-place, you don't need to return my_string.